UIKitTests/UIApplication+NSNotificationTests.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  UIApplication+NSNotificationTests.swift
+//  UIKitTests
+//
+//  Created by Chetan Agarwal on 30.11.18.
+//  Copyright © 2018 flowkey. All rights reserved.
+//
+
+import XCTest
+@testable import UIKit
+
+class UIApplication_NSNotificationTests: XCTestCase {
+    func testApplicationPostsWillEnterForegroundNotification() {
+        let notificationExpectation = expectation(forNotification: .UIApplicationWillEnterForeground,
+                                                  object: nil,
+                                                  handler: nil)
+        UIApplication.onWillEnterForeground()
+        wait(for: [notificationExpectation], timeout: 0.1)
+    }
+
+    func testApplicationPostsDidBecomeActiveNotification() {
+        let notificationExpectation = expectation(forNotification: .UIApplicationDidBecomeActive,
+                                                  object: nil,
+                                                  handler: nil)
+        UIApplication.onDidEnterForeground()
+        wait(for: [notificationExpectation], timeout: 0.1)
+    }
+
+    func testApplicationPostsWillResignActiveNotification() {
+        let notificationExpectation = expectation(forNotification: .UIApplicationWillResignActive,
+                                                  object: nil,
+                                                  handler: nil)
+        UIApplication.onWillEnterBackground()
+        wait(for: [notificationExpectation], timeout: 0.1)
+    }
+
+    func testApplicationPostsDidEnterBackgroundNotification() {
+        let notificationExpectation = expectation(forNotification: .UIApplicationDidEnterBackground,
+                                                  object: nil,
+                                                  handler: nil)
+        UIApplication.onDidEnterBackground()
+        wait(for: [notificationExpectation], timeout: 0.1)
```

Comment:
minor nitpick: I thought there was a `wait(for: Expectation)` API that took a single expectation rather than an array? That may be cleaner here. Not a big deal either way though

Reply:
The `XCTWaiter` class doesn't define a method to wait for a single expectation.
It's possible to write an `extension` and support this, however in this particular test case - it's also possible to skip assigning the variable altogether and use the function `waitForExpectations(timeout: ...)` - I think is very simple test cases like these - this could be a good approach.

Reply:
Oh yeah, cool. That sounds good, I guess even if we expand the tests with more expectations we probably won't need to wait for each expectation with a different timeout etc.